Sources/Dependencies/DependencyValues.swift

```diff
@@ -334,3 +334,47 @@ private let defaultContext: DependencyContext = {
     return .live
   }
 }()
+
+extension TestDependencyKey {
+  static var stableOpenedLiveValue: Value? {
+    let id = DefaultValueID(key: Self.self, context: .live)
+    defaultValueStorageLock.lock()
+    defer { defaultValueStorageLock.unlock() }
+    if let value = defaultValueStorage[id] as? Value { return value }
+    let value = _liveValue(Self.self) as? Value
+    defaultValueStorage[id] = value
+    return value
+  }
+  
+  static var stableTestValue: Value {
+    let id = DefaultValueID(key: Self.self, context: .test)
+    defaultValueStorageLock.lock()
+    defer { defaultValueStorageLock.unlock() }
+    if let value = defaultValueStorage[id] as? Value { return value }
+    let value = self.testValue
+    defaultValueStorage[id] = value
+    return value
+  }
+  
+  static var stablePreviewValue: Value {
+    let id = DefaultValueID(key: Self.self, context: .preview)
+    defaultValueStorageLock.lock()
+    defer { defaultValueStorageLock.unlock() }
+    if let value = defaultValueStorage[id] as? Value { return value }
+    let value = self.previewValue
+    defaultValueStorage[id] = value
+    return value    
+  }
+}
```

Comment:
There's a fair amount of repetition, but I don't think it would be much nicer to funnel everyone through a common helper function with a force-unwrap and an autoclosure. But I can perform the change if needed.